gen/enum.go

```diff
@@ -190,6 +191,26 @@ func enum(g Generator, spec *compile.EnumSpec) error {
 			return nil
 		}
 
+		<$sr := newVar "sr">
+		// Decode reads off the encoded <$enumName> directly off of the wire.
+		//
+		//   sReader := BinaryStreamer.Reader(reader)
+		//
+		//   var <$v> <$enumName>
+		//   if err := <$v>.Decode(sReader); err != nil {
+		//     return <$enumName>(0), err
+		//   }
+		//   return <$v>, nil
+		func (<$v> *<$enumName>) Decode(<$sr> <$stream>.Reader) error {
+			<- $rVal := newVar "rVal" ->
```

Comment:
optional nitty nit: I would have just named this "x", "i", or "val". There's not much ambiguity here; it's a really small function to require the "rVal" disambiguation.